### ISSUE TITLE
Cannot start UserTransactions from standalone clients

### ISSUE THREAD

**Issue by riksweeney** [payara/Payara#4322]
I'm in the process of moving from Glassfish 4. We have some standalone programs which interact with Glassfish and make use of the UserTransaction. However, when attempting to run them on Payara, I encounter the following error when starting the transaction:

```
java.lang.IllegalStateException: Service fish.payara.nucleus.requesttracing.RequestTracingService was started at level -2 but it has a run level of 10.  The full descriptor is SystemDescriptor(
	implementation=fish.payara.nucleus.requesttracing.RequestTracingService
	name=requesttracing-service
	contracts={fish.payara.nucleus.requesttracing.RequestTracingService,org.jvnet.hk2.config.ConfigListener,fish.payara.monitoring.collect.MonitoringDataSource}
	scope=org.glassfish.hk2.runlevel.RunLevel
	qualifiers={}
	descriptorType=CLASS
	descriptorVisibility=NORMAL
	metadata=runLevelValue={10}
	rank=0
	loader=org.glassfish.hk2.bootstrap.impl.Hk2LoaderPopulatorPostProcessor$1@1551677a
	proxiable=null
	proxyForSameScope=null
	analysisName=null
	id=615
	locatorId=0
	identityHashCode=221518168
	reified=true)
	at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.validate(AsyncRunLevelContext.java:446)
	at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:299)
	at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:89)
	at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2126)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetService(ServiceLocatorImpl.java:777)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getUnqualifiedService(ServiceLocatorImpl.java:789)
	at org.jvnet.hk2.internal.IterableProviderImpl.get(IterableProviderImpl.java:111)
	at com.sun.enterprise.transaction.JavaEETransactionManagerSimplified.getRequestTracing(JavaEETransactionManagerSimplified.java:201)
	at com.sun.enterprise.transaction.JavaEETransactionManagerSimplified.initJavaEETransaction(JavaEETransactionManagerSimplified.java:635)
	at com.sun.enterprise.transaction.JavaEETransactionManagerSimplified.begin(JavaEETransactionManagerSimplified.java:887)
	at com.sun.enterprise.transaction.JavaEETransactionManagerSimplified.begin(JavaEETransactionManagerSimplified.java:851)
	at com.sun.enterprise.transaction.UserTransactionImpl.begin(UserTransactionImpl.java:169)
```

Here is a very simple test case to reproduce the issue

```
import javax.naming.Context;
import javax.naming.InitialContext;
import javax.transaction.UserTransaction;

public class TransactionTest {

    public static void main (String [] args) {

        UserTransaction tx = null;

        try {
			System.out.println ("Getting context");

            final Context context = new InitialContext ();

            System.out.println ("Getting UserTransaction");

            tx = (UserTransaction) context.lookup ("java:comp/UserTransaction");

            System.out.println ("Setting transaction timeout");

            tx.setTransactionTimeout (15);

            System.out.println ("Beginning transaction");

            tx.begin ();

            System.out.println ("Rolling back transaction");

            tx.rollback ();
        } catch (final Exception e) {
            e.printStackTrace ();
        }
    }
}
```

Either something has changed or I was never doing it right to begin with. Any help or suggestions would be appreciated.

Richard

**Comment by smillidge** [comment]
Is this error on the client or server?

**Comment by riksweeney** [comment]
Sorry, this error is on the client. I downloaded payara-5.193.1 and started it up using

asadmin start-domain

Once the domain started up, I compiled and ran the test code above which produced the stacktrace.

**Comment by smillidge** [comment]
Looking at the code I suspect this is a bug as the transaction service is trying to start the request tracing service.

**Comment by riksweeney** [comment]
Ah OK. Is there a way to prevent the request tracing service from starting up or outright disable it?

**Comment by fturizo** [comment]
@riksweeney, this is a known issue that was introduced in release *5.193.1* and is tracked internally by the ID `CUSTCOM-70`. You can expect a fix to this issue to be included in the next public release.

**Comment by riksweeney** [comment]
Great, thank you!

**Comment by OndroMih** [comment]
This is now fixed in the latest release.

**Comment by riksweeney** [comment]
Was this issue only fixed in the Enterprise edition? The issue is still present in the Community edition (5.2020.3).

**Comment by OndroMih** [comment]
Hi @riksweeney, we fixed a similar problem in version 5.201, which is a community version. It's possible that the fix didn't entirely fix your usecase. Can you please give us a simple reproducer project so that we can verify that it still fails with 5.2020.3?

**Comment by OndroMih** [comment]
Closing now as this issue is now reopened as #4809.
